alerting/src/main/kotlin/org/opensearch/alerting/service/DeleteMonitorService.kt

```diff
@@ -95,57 +95,69 @@ object DeleteMonitorService :
 
     private suspend fun deleteDocLevelMonitorQueriesAndIndices(monitor: Monitor) {
         try {
-            val metadata = MonitorMetadataService.getMetadata(monitor)
-            metadata?.sourceToQueryIndexMapping?.forEach { (_, queryIndex) ->
+            if (monitor.owner == "alerting") {
```

Comment:
plz delete based on a boolean flag that defaults to false. 
This is an anti pattern and makes code unmaintainable

Reply:
fixed it based on boolean flag passed from `common-utils`.